packages/fxa-payments-server/src/components/Portal.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { memo, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+
+import './Portal.scss';
+
+type PortalProps = {
+  id: string,
+  children: React.ReactNode
+};
+
+const Portal = ({
```

Comment:
[Portals](https://reactjs.org/docs/portals.html) in React are DOM elements outside of the normal rendering tree. (e.g. a DOM element at the root of the document that contains dialog panes.) This component helps ensure one DOM parent with the given ID exists when needed, and then removed when the last child is gone.